src/ln/peer_handler.rs

```diff
@@ -516,10 +520,36 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 														self.initial_syncs_sent.fetch_add(1, Ordering::AcqRel);
 														local_features.set_initial_routing_sync();
 													}
+													//we need to let them know about us, before we can send stuff to them
 													encode_and_send_msg!(msgs::Init {
 														global_features: msgs::GlobalFeatures::new(),
 														local_features,
 													}, 16);
+													if self.initial_syncs_sent.load(Ordering::Acquire) < INITIAL_SYNCS_TO_SEND {
```

Comment:
We shouldn't be dumping our routing table at the first N nodes we connect to? We should only do this when *their* init asks for it.